### PR TITLE
Rename DUSD loops to dusd-vault attribute key

### PR DESCRIPTION
**Pull request by sieniven**
## Summary

- Name changes for gov attributes for the approved DFIP on DUSD staking (https://github.com/DeFiCh/dfips/issues/281), with new vaults attribute with key id dusd-vault, and vault key dusd-vault-enabled.

## Implications

- Storage
  - [ ] Database reindex required
  - [ ] Database reindex optional
  - [ ] Database reindex not required
  - [x] None

- Consensus
  - [ ] Network upgrade required
  - [ ] Includes backward compatible changes
  - [ ] Includes consensus workarounds
  - [ ] Includes consensus refactors
  - [x] None

### CODE REVIEW

File: src/masternodes/govvariables/attributes.h [hunk 1]
```diff
@@ -93,7 +98,7 @@ enum DFIPKeys : uint8_t {
     MintTokens           = 't',
     EVMEnabled           = 'u',
     ICXEnabled           = 'v',
-    AllowDUSDLoops       = 'w',
+    DUSDVault            = 'w',
```
Comment (Bushstar) [inline src/masternodes/govvariables/attributes.h]: Should probably have its own entry under VaultKeys. DFIPKeys are used for AttributeTypes::Param. That's how we would normally add a new one to scale for additional uses later.

Comment (sieniven) [reply src/masternodes/govvariables/attributes.h]: Sure will make the changes, i thought of leaving it here since the implementation is a special DFIP.

Comment (sieniven) [reply src/masternodes/govvariables/attributes.h]: Resolved in commit [7b343b7](https://github.com/DeFiCh/ain/pull/2187/commits/7b343b7661a5e7faa810002a90fc3d91806d991f)

File: test/functional/feature_dusd_loans.py [hunk 2]
```diff
@@ -535,30 +535,21 @@ def check_looped_dusd(self):
...
-        self.goto_next_height()
-
         # min coll ratio still applies
         assert_raises_rpc_error(-32600,
                                 "Vault does not have enough collateralization ratio defined by loan scheme - 148 < 150",
                                 self.takeloan_withdraw, vault_id, "67.00000000@DUSD", 'takeloan')
 
-
+        self.goto_next_height()
+        self.nodes[0].setgov({"ATTRIBUTES": {'v0/vaults/dusd-vault/dusd-vault-enabled': 'true'}})
```
Comment (prasannavl) [inline test/functional/feature_dusd_loans.py:544]: Either `v0/vaults/dusd-vault` as bool, or `v0/vaults/dusd-vault/enabled` if we have to accommodate for future unknowns.

Comment (sieniven) [reply test/functional/feature_dusd_loans.py]: Resolved in commit 2496058

File: src/masternodes/govvariables/attributes.cpp [hunk 3]
```diff
@@ -1611,6 +1653,14 @@ Res ATTRIBUTES::Validate(const CCustomCSView &view) const {
                 }
                 break;
 
+            case AttributeTypes::Vaults:
+                if (attrV0->typeId == VaultIDs::DUSDVault && attrV0->key == VaultKeys::DUSDVaultEnabled) {
+                    if (view.GetLastHeight() < Params().GetConsensus().NextNetworkUpgradeHeight) {
+                        return Res::Err("Cannot be set before NextNetworkUpgrade");
```
Comment (prasannavl) [inline src/masternodes/govvariables/attributes.cpp:1659]: In a separate PR, can we pull all of these NextNetworkUpgrade strings, and collect it under DeFiErrors? Once we name the upgrade, should also help to just make a single change that chase after all these strings.

**Review by prasannavl — APPROVED** [review]

**Comment by prasannavl** [comment]
https://github.com/DeFiCh/ain/pull/2187#discussion_r1267509523

> In a separate PR, can we pull all of these NextNetworkUpgrade strings, and collect it under DeFiErrors? Once we name the upgrade, should also help to just make a single change that chase after all these strings.

**Pull request merged**